jax/_src/random.py

```diff
@@ -926,21 +949,42 @@ def _next_kxv(kxv):
     return key, X, V, U
 
   # initial state is chosen such that _cond_fn will return True
+  key, subkey = _split(key)
+  u_boost = uniform(subkey, (), dtype=dtype)
   _, _, V, _ = lax.while_loop(_cond_fn, _body_fn, (key, zero, one, _lax_const(alpha, 2)))
-  z = lax.mul(lax.mul(d, V), boost)
-  return lax.select(lax.eq(z, zero), jnp.finfo(z.dtype).tiny, z)
+  if log_space:
+    # TODO(jakevdp): there are negative infinities here due to issues mentioned above. How should
+    # we handle those?
+    log_boost = lax.select(boost_mask, zero, lax.mul(lax.log(u_boost), lax.div(one, alpha_orig)))
+    return lax.add(lax.add(lax.log(d), lax.log(V)), log_boost)
```

Comment:
I think it is a bit inconvenient to return loggamma samples that after taking exponent, we get 0. Maybe to be comparable with `gamma` sampler, we can do
```
z = lax.add(lax.add(lax.log(d), lax.log(V)), log_boost)
log_tiny = lax.log(jnp.finfo(z.dtype).tiny)
return lax.select(lax.lt(z, log_tiny), log_tiny, z)
```
Beside convenient for users, this way we might avoid -inf samples, avoid special treatment for small values in loggamma grad, avoid nan in grad test, and make gamma sampler = exp(loggamma sampler) even for small values.

Reply:
I'm going to push back here:  we don't want log-gamma to be comparable to gamma: we want it to be able to represent values that gamma cannot. The primary motivation for adding the log-gamma sampler is to be able to represent values that exponentiate to zero. If we truncate the output at `log(tiny)`, we will lose the ability to robustly sample `beta` and `dirichlet` values for small alpha.

Reply:
That's a good point! Thanks for the explanation.